### ISSUE TITLE
[microsite] Plugin Marketplace - Category Filter

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#13014]
## Feature Suggestion

It would be great now that there are many items in the Plugin Marketplace that we add a filter by category so that way you can find plugins a little more easily

## Possible Implementation

This could simply be a drop down with all the available categories or as complex as a multi-select and then the cards would filter based on the selection

## Context

Now that there are lots of awesome plugins it's getting a bit harder to find things

**Comment by jhaals** [comment]
Yes that sounds like a great idea. Would it be worth doing this as a followup after Docusaurus 2?

**Comment by awanlin** [comment]
Yes, I gave this a try myself while I was on vacation and had the urge to code a little. I could get the drop down populated with the categories but trying to get it to filter just wasn't going to work

**Comment by Rugvip** [comment]
Tbh it might be a good idea to put this one on ice until #12019 is through. Client-side scripting is a bit of a pain on the microsite currently.

**Comment by awanlin** [comment]
Still something that would be great to have once #12019 has been completed 👍

**Comment by awanlin** [comment]
Still want this ⬆️

**Comment by freben** [comment]
And likely more possible now that #12019 is in!

**Comment by antoniobergas** [comment]
@awanlin you already gave a try right? Want an extra hand?

**Comment by awanlin** [comment]
Yeah, now that we are on Docusaurus 2 this should be much easier. My workload is pretty high right now - I already have 4 bigger PRs waiting for review - so feel free to run with this @antoniobergas 🚀

**Comment by awanlin** [comment]
Yes, that sounds awesome!

**Comment by awanlin** [comment]
Stretch goal: multi-select if that's not too hard

**Comment by antoniobergas** [comment]
@awanlin

So wanna give some background about what I did.

Implement a single selector was easy so then I tried the multi one but the feeling in the UI was not good enough from my point of view (If so select more than 2 categories the rest was hidden inside the selector)

So after make a quick search I decide to go for a chip filter. I honestly love the final result and I think that fits more the the microsite design.

Let me know what you think! 😊

PD: By the way I don't have that much expertise on React so further suggestions will be appreciated to make a better code!
